### ISSUE TITLE
Train on own data with single class

### ISSUE THREAD

**Issue by dlyubomirov** [jwyang/faster-rcnn.pytorch#480]
I want to build a real-time face detector, so my idea was to load a pre-trained network with a ResNet101-backbone and then train it with my own annotated image database, where in every picture a person's face is annotated.

My idea is to remove all classes besides _background_ and add a new class called _face_. This means I obviously have to modify the last layer. What I don't understand is how I can train on my own data.

I followed the advice given in #221 and I tried to store my image dataset in a PASCAL VOC format, so that I can use the already implemented class Pascal Voc. I also generated xml-files, where the bounding box coordinates are stored. However I get some really strange bugs and I think I have to further modify the code to get it running on my own data. Since I am relatively unexperienced in neural network programming, I am having a hard time and I don't know where to start.

Can someone tell me how I can train on my own data?

**Comment by EMCP** [comment]
sounds like you need to either check your xml, images, dataset itself, or post the bugs here

**Comment by devendraswamy** [comment]
Actually I have my own dataset in format of ["filename" "boxes label values Like x-min y-min x-max y-max" "class-name"] and filename consist the path of the image, so how can given this dataset to model for testing and training , please  help me ..

**Comment by stadtgiraffe** [comment]
@devendraswamy I have the exact same issue. The advice given in #221 didn't help either. Isn´t there a easy way to use faster rcnn for just ["filename", "x1","y1","x2","y2","class-name"] annotatione? If anyone knows I`d be happy to hear some suggestions. Thank you in advance!

**Comment by EMCP** [comment]
@stadtgiraffe that would be very very painful the second you have images with multiple objects.. I suggest you try building your dataset in coco with something like https://github.com/jsbroks/coco-annotator , set this project to coco , and try again

**Comment by devendraswamy** [comment]
I already had a dataset, but I encrypt that data in the form of .txt only, here I want to access the data from that .txt file because my data not in the form of coco or pascal, so please give me any suggestion  

Thanking you!

**Comment by EMCP** [comment]
not sure what you mean by encrypting data into a text file.

if you dont want to use the data loaders provided, you'd likely need to interface your own.. I cannot recommend that since I haven't tried it, but it is likely possible 

You'd likely craft one based on the others, then pick it up via commandline arguements

https://github.com/jwyang/faster-rcnn.pytorch/blob/master/trainval_net.py#L41-L43

again, don't recommend it but if you need to.. it is possible

**Comment by devendraswamy** [comment]
Thank you for your valuable information, one more doubt please help me . i want to increase the model accuracy , now  its giving good results that OK but getting better accuracy , which parameters i have to tune ( tell me about parameter tuning).

Thanking you!

**Comment by EMCP** [comment]
you likely need to review your data, not just the hyper parameters... for the latter, I suggest the following

https://www.coursera.org/learn/deep-neural-network

**Comment by devendraswamy** [comment]
Thank you very much for your support , could you please tell me about how can i decrease the loss value in this project ?? kindly give me the suggestion.

**Comment by EMCP** [comment]
there are too many factors in order for me to know that... you need to review your data, review the parameters... this is the challenge of ML projects quite frankly.

if you can, use the tensorboard https://github.com/lanpa/tensorboardX

**Comment by devendraswamy** [comment]
Thank you so much for your support !

**Comment by devendraswamy** [comment]
sir i successfully run tensorboardx and  one more doubt , i want to check the object detection for coco dataset but demo.py is only applicable for pascal voc dataset, could you please suggest the solution for testing the 80 classes of coco with demo.py code, in this demo.py which lines i have to modify ??

Thanking you !

**Comment by devendraswamy** [comment]
Sir i solved this and i modified the demp.py code , thanking you.

**Comment by devendraswamy** [comment]
sir any suggestions for data augmentation ??
